First pass at the Iron Irwini release notes.

clalancette opened the pull request:
That is, add in the full changelog, and also greatly expand the release notes themselves.

I should point out that the full changelog is necessarily *not* complete; it only contains information on things that have already been released.

The release note page is a cut-down version of the full changelog that just has things that might be interesting to end users.  What is currently in there was my opinion, so this list may be expanded or reduced based on thoughts from other people.

In both cases, we should be able to iteratively add new items here as they are landed.

clalancette committed:
First pass at the Iron Irwini release notes.

That is, add in the full changelog, and also greatly
expand the release notes themselves.

I should point out that the full changelog is necessarily
*not* complete; it only contains information on things that
have already been released.

The release note page is a cut-down version of the full
changelog that just has things that might be interesting
to end users.  What is currently in there was my opinion,
so this list may be expanded or reduced based on thoughts
from other people.

In both cases, we should be able to iteratively add new
items here as they are landed.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

gavanderhoorn commented on source/Releases/Release-Iron-Irwini.rst:
```diff
@@ -85,6 +273,266 @@ Later we would like to have support for a full configuration file (see: https://
+"""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+Specifying a QoS of KEEP_LAST with a depth of 0 is a nonsensical arrangement, since the entity wouldn't be able to send or receive any data.
+``rclpy`` will now print a warning if this combination is specified, but will still continue on and let the underlying middleware choose a sane value (generally a depth of 1).
```
```suggestion
``rclcpp`` will now print a warning if this combination is specified, but will still continue on and let the underlying middleware choose a sane value (generally a depth of 1).
```

clalancette replied:
Whoops, good catch.  Copy-n-paste error :).

christophebedard commented:
I feel like the list of contributors under each package doesn't really match the list of contributors in the package's CHANGELOG file. For example, the aggregated release notes for `tracetools` combine the release notes from 5.0.0 and 5.1.0 but only use the contributors from 5.0.0.

clalancette committed:
Update source/Releases/Release-Iron-Irwini.rst

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

Co-authored-by: G.A. vd. Hoorn <g.a.vanderhoorn@tudelft.nl>

clalancette commented:
> I feel like the list of contributors under each package doesn't really match the list of contributors in the package's CHANGELOG file. For example, the aggregated release notes for `tracetools` combine the release notes from 5.0.0 and 5.1.0 but only use the contributors from 5.0.0.

Huh, good point.  There must be a bug in the script I use to generate this; I'll take a look.

fujitatomoya reviewed (commented):
@clalancette I will add the note for

- Connect/Disconnect event support, https://github.com/ros2/rmw/issues/330
- External logging level reconfiguration, https://github.com/ros2/ros2/issues/1355 (W.I.P)

i save the reminders for each issue about this task not to miss.

clalancette commented:
> i save the reminders for each issue about this task not to miss.

Great, thanks @fujitatomoya .

clalancette committed:
Updates from the latest script fixes.

As pointed out by Christophe Bedard, there were some
people missing from the Contributors.  After doing some
work on the script in https://github.com/clalancette/ros_changelog_from_pkgs,
this is the update.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

clalancette commented:
> I feel like the list of contributors under each package doesn't really match the list of contributors in the package's CHANGELOG file. For example, the aggregated release notes for `tracetools` combine the release notes from 5.0.0 and 5.1.0 but only use the contributors from 5.0.0.

@christophebedard Based on this comment, I fixed some bugs in the script I use to generate this (https://github.com/clalancette/ros_changelog_from_pkgs).  The latest commit adds in the new output, and at least for the repositories I spot-checked, added in the missing Contributors.  Tomorrow I'll do a more thorough check, but thanks for diligently checking!

clalancette commented:
All right.  I didn't exhaustively go over every package (there are 350 of them), but I did a more thorough spot-check and the contributor list looks good on every package I checked.  So I'm pretty happy with this.

Yadunund commented on source/Releases/Release-Iron-Irwini.rst:
```diff
@@ -21,15 +19,18 @@ Iron Irwini is primarily supported on the following platforms:
 Tier 3 platforms:
 
-TBD
+* Ubuntu 20.04 (Focal): ``amd64``
```
Looking at the [REP 2000](https://www.ros.org/reps/rep-2000.html#iron-irwini-may-2023-november-2024) we no longer plan to support `focal` at Tier 3.

It however mentions `OpenEmbedded` at Tier 3. Do we want to add the `arm32` options too?

clalancette replied:
> Looking at the [REP 2000](https://www.ros.org/reps/rep-2000.html#iron-irwini-may-2023-november-2024) we no longer plan to support `focal` at Tier 3.

Good call, I'll remove this.

> It however mentions OpenEmbedded at Tier 3. Do we want to add the arm32 options too?

I'm somewhat reluctant to do that, as we have no documentation whatsoever on this site about OpenEmbedded (and we've never added it before).  So if you don't mind, I'd prefer to leave it out.

clalancette replied:
And fixed in b59226e.

clalancette committed:
Fixes from review.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

Yadunund commented on source/Releases/Release-Iron-Irwini.rst:
```diff
@@ -85,6 +272,266 @@ Later we would like to have support for a full configuration file (see: https://
+It is now possible to force the ROS parameter file parser to interpret a field as a string using the YAML ``!!str`` syntax.
+See https://github.com/ros2/rcl/pull/999 for more information.
+
+``rclcpp``
```
Do we want to combine this `rclcpp` feature with the `rclcpp` section in line 145?

clalancette replied:
I put new features under the "New in this release" and fixes/updates under the "Changes since the previous release" heading.  That's why there is the duplication.

Maybe that distinction doesn't matter, and we should just have a single "Changes since the previous release" heading, and then combine the sections.  I don't have a strong opinion either way; what do you think?

Yadunund replied:
Ah gotcha thanks for the clarification. We can leave it as is

Yadunund commented on source/Releases/Release-Iron-Irwini.rst:
```diff
@@ -85,6 +272,266 @@ Later we would like to have support for a full configuration file (see: https://
+
+See https://github.com/ros2/rclcpp/pull/2048 for more information.
+
+``rclpy``
```
Same as above. Combine with `rclpy` first mentioned in line 180?

Yadunund reviewed (commented):
Final set of comments from my side!

Yadunund reviewed (approved):
LGTM! Thanks for putting this together 🙇🏼
